tests/functions.tests

```diff
@@ -51,6 +51,20 @@ fun (x) {x:a} : a
 stderr : @.*
 exit : 1
 
+Infers effect variables
+fun (f, x: Int) { f(x) + 0 }
+stdout : @fun : \(\(Int\) -a-> Int, Int\) -a-> Int
```

Comment:
I'm not sure why we want to show the variable here. In combination with the next test, it seems that adding the type annotation (without the variable) suppresses the printing of the variable. This seems  inconsistent to me.

Reply:
This test runs without the setting enabled, just to check that the previous behaviour is maintained - hence it displaying the type variables.

Reply:
Yes, I meant that we should have a test for the following: If the setting is enabled and the effects are not the same, the printer must not use plain `->`